tests/serial/time/implicit/parallel-coupling/ReadWriteScalarDataWithWaveformSubcyclingVaryingTimeGrids.cpp

```diff
@@ -0,0 +1,114 @@
+#ifndef PRECICE_NO_MPI
+
+#include "testing/Testing.hpp"
+
+#include <precice/precice.hpp>
+#include <vector>
+
+using namespace precice;
+
+BOOST_AUTO_TEST_SUITE(Integration)
+BOOST_AUTO_TEST_SUITE(Serial)
+BOOST_AUTO_TEST_SUITE(Time)
+BOOST_AUTO_TEST_SUITE(Implicit)
+BOOST_AUTO_TEST_SUITE(ParallelCoupling)
+
+/**
+ * @brief Test to run a simple coupling with time grids that change with the waveform iteration and subcycling.
+ *
+ * Provides a dt argument to the read function. A first order waveform is used.
+ */
+BOOST_AUTO_TEST_CASE(ReadWriteScalarDataWithWaveformSubcyclingVaryingTimeGrids)
+{
+  PRECICE_TEST("SolverOne"_on(1_rank), "SolverTwo"_on(1_rank));
+
+  Participant precice(context.name, context.config(), 0, 1);
+
+  typedef double (*DataFunction)(double);
+
+  DataFunction dataOneFunction = [](double t) -> double {
+    return (double) (2 + t);
+  };
+  DataFunction dataTwoFunction = [](double t) -> double {
+    return (double) (10 + t);
+  };
+  DataFunction writeFunction;
+  DataFunction readFunction;
+
+  std::string meshName, writeDataName, readDataName;
+  if (context.isNamed("SolverOne")) {
+    meshName      = "MeshOne";
+    writeDataName = "DataOne";
+    writeFunction = dataOneFunction;
+    readDataName  = "DataTwo";
+    readFunction  = dataTwoFunction;
+  } else {
+    BOOST_TEST(context.isNamed("SolverTwo"));
+    meshName      = "MeshTwo";
+    writeDataName = "DataTwo";
+    writeFunction = dataTwoFunction;
+    readDataName  = "DataOne";
+    readFunction  = dataOneFunction;
+  }
+
+  double   writeData, readData;
+  VertexID vertexID;
+
+  double v0[] = {0, 0, 0};
+  vertexID    = precice.setMeshVertex(meshName, v0);
+  int    timestepCheckpoint;
+  double time = 0;
+
+  if (precice.requiresInitialData()) {
+    writeData = writeFunction(time);
+    precice.writeData(meshName, writeDataName, {&vertexID, 1}, {&writeData, 1});
+  }
+
+  precice.initialize();
+  double time_window_size = precice.getMaxTimeStepSize();
+  double maxDt            = time_window_size;
```

Comment:
```suggestion
  double timeWindowSize = precice.getMaxTimeStepSize();
  double maxDt            = timeWindowSize;
```
Please use `camelCase`, not `snake_case`. @fsimonis Interesting that our linter does not fix this...